Source/WebGPU/WebGPU/CommandEncoder.swift

```diff
@@ -743,4 +748,186 @@ extension WebGPU.CommandEncoder {
         }
         return true
     }
+
+    public func copyTextureToTexture(source: WGPUImageCopyTexture, destination: WGPUImageCopyTexture, copySize: WGPUExtent3D) {
+        guard source.nextInChain != nil, destination.nextInChain != nil else {
+            return
+        }
+
+        // https://gpuweb.github.io/gpuweb/#dom-gpucommandencoder-copytexturetotexture
+
+        guard  self.prepareTheEncoderState() else  {
+            self.generateInvalidEncoderStateError()
+            return
+        }
+        let error = self.errorValidatingCopyTextureToTexture(source, destination, copySize)
+        guard error != nil else {
```

Comment:
shouldn't this be ==

Reply:
It should be.. I haven't even self reviewed this yet...or tested this yet.. I just pushed to try on a different machine.. I am getting a linker error at the moment.. 
Yes I probably should have made this draft.. my bad..